numpy/lib/tests/test_function_base.py

```diff
@@ -4325,6 +4326,22 @@ def test_float16_gh_29003(self):
         assert value.dtype == np.float16
 
 
+@hypothesis.given(
+    a=arrays(
+        # real_dtype online in array_api, not in numpy
+        dtype=make_strategies_namespace(np).real_dtypes(),
+        shape=array_shapes(max_dims=1, min_side=1),
+    ).filter(lambda a: np.all(np.isfinite(a))),
+    q=st.floats(min_value=0, max_value=1),
+)
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
+def test_quantile_properties(a, q):
```

Comment:
Should this not be under the `TestQuantile` class? Also, perhaps a more descriptive name would be better (maybe `test_quantile_within_bounds`)?

Reply:
> Should this not be under the `TestQuantile` class?

I don't know, but I can put it there. The function doesn't depend on any attributes and I see no application for inheritance, so I thought a free/plain function is more idiomatic than a method.


> Also, perhaps a more descriptive name would be better (maybe `test_quantile_within_bounds`)?

If it stays like this, I agree. I planned on adding the type checks here (and any other property we agree on) and extending the test to cover all `method`s.

Alternatively, we could write multiple short functions, but it seemed more efficient to me, to scan all inputs only once.